src/plugins/image_embeddable/public/plugin.ts

```diff
@@ -42,36 +43,42 @@ export interface SetupContract {}
 export interface StartContract {}
 
 export class ImageEmbeddablePlugin
-  implements Plugin<SetupContract, StartContract, SetupDependencies, StartDependencies>
+  implements
+    Plugin<
+      SetupContract,
+      StartContract,
+      ImageEmbeddableSetupDependencies,
+      ImageEmbeddableStartDependencies
+    >
 {
-  constructor(protected readonly context: PluginInitializerContext) {}
-
-  public setup(core: CoreSetup<StartDependencies>, plugins: SetupDependencies): SetupContract {
-    const start = createStartServicesGetter(core.getStartServices);
-    plugins.embeddable.registerEmbeddableFactory(
-      IMAGE_EMBEDDABLE_TYPE,
-      new ImageEmbeddableFactoryDefinition({
-        start: () => ({
-          application: start().core.application,
-          overlays: start().core.overlays,
-          files: start().plugins.files.filesClientFactory.asUnscoped(),
-          externalUrl: start().core.http.externalUrl,
-          theme: start().core.theme,
-          getUser: async () => {
-            const security = start().plugins.security;
-            return security ? await security.authc.getCurrentUser() : undefined;
-          },
-          uiActions: start().plugins.uiActions,
-          isScreenshotMode: () => plugins.screenshotMode?.isScreenshotMode() ?? false,
-        }),
-      })
-    );
+  constructor() {}
 
+  public setup(
+    core: CoreSetup<ImageEmbeddableStartDependencies>,
+    plugins: ImageEmbeddableSetupDependencies
+  ): SetupContract {
     plugins.uiActions.registerTrigger(imageClickTrigger);
     return {};
   }
 
-  public start(core: CoreStart, plugins: StartDependencies): StartContract {
+  public async start(
+    core: CoreStart,
+    plugins: ImageEmbeddableStartDependencies
+  ): Promise<StartContract> {
+    setKibanaServices(core, plugins);
+
+    await untilPluginStartServicesReady();
+
+    registerReactEmbeddableFactory(IMAGE_EMBEDDABLE_TYPE, async () => {
+      const { getImageEmbeddableFactory } = await import(
+        './image_embeddable/get_image_embeddable_factory'
+      );
+      return getImageEmbeddableFactory({ embeddableEnhanced: plugins.embeddableEnhanced });
+    });
+
+    const { registerCreateImageAction } = await import('./actions/create_image_action');
+    registerCreateImageAction();
```

Comment:
The async import here stops the bundle size from being blown up by the `PresentationPublishing` package